// carbon-components-react/es/components/UIShell/SideNavSwitcher.js
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react'; // TODO: sync with @carbon/icons-react

var ChevronDown = function ChevronDown() {
  return React.createElement("svg", {
    "aria-hidden": "true",
    width: "20",
    height: "20",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 32 32"
  }, React.createElement("path", {
    d: "M16 22L6 12l1.414-1.414L16 19.172l8.586-8.586L26 12 16 22z"
  }));
};

var SideNavSwitcher = React.forwardRef(function (props, ref) {
  var customClassName = props.className,
      labelText = props.labelText,
      onChange = props.onChange,
      options = props.options;
  var className = cx('bx--side-nav__switcher', customClassName); // Note for usage around `onBlur`: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md

  return React.createElement("div", {
    className: className
  }, React.createElement("label", {
    htmlFor: "side-nav-switcher",
    className: "bx--assistive-text"
  }, labelText), React.createElement("select", {
    id: "carbon-side-nav-switcher",
    className: "bx--side-nav__select",
    defaultValue: "",
    onBlur: onChange,
    onChange: onChange,
    ref: ref
  }, React.createElement("option", {
    className: "bx--side-nav__option",
    disabled: true,
    hidden: true,
    value: ""
  }, labelText), options.map(function (option) {
    return React.createElement("option", {
      key: option,
      className: "bx--side-nav__option",
      value: option
    }, option);
  })), React.createElement("div", {
    className: "bx--side-nav__switcher-chevron"
  }, React.createElement(ChevronDown, null)));
});
SideNavSwitcher.propTypes = {
  /**
   * Provide an optional class to be applied to the containing node
   */
  className: PropTypes.string,

  /**
   * Provide the label for the switcher. This will be the firt visible option
   * when someone views this control
   */
  labelText: PropTypes.string.isRequired,

  /**
   * Provide a callback function that is called whenever the switcher value is
   * updated
   */
  onChange: PropTypes.func,

  /**
   * Provide an array of options to be rendered in the switcher as an
   * `<option>`. The text value will be what is displayed to the user and is set
   * as the `value` prop for each `<option>`.
   */
  options: PropTypes.arrayOf(PropTypes.string).isRequired
};
export default SideNavSwitcher;
