// carbon-components-react/lib/components/UIShell/classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sidenav = exports.truncate = exports.global = exports.name = exports.navigation = exports.button = exports.action = exports.header = void 0;
var header = {
  header: 'bx--header'
};
exports.header = header;
var action = {
  action: 'bx--header__action',
  state: {
    active: 'bx--header__action--active'
  }
};
exports.action = action;
var button = {
  button: 'bx--header__menu-trigger'
};
exports.button = button;
var navigation = {
  navigation: 'bx--header__nav',
  menu: 'bx--header__menu',
  menubar: 'bx--header__menu-bar',
  menuitem: 'bx--header__menu-item',
  menutitle: 'bx--header__menu-title',
  submenu: 'bx--header__submenu',
  menuarrow: 'bx--header__menu-arrow'
};
exports.navigation = navigation;
var name = {
  name: 'bx--header__name',
  platform: 'bx--header__platform-name'
};
exports.name = name;
var global = {
  global: 'bx--header__global'
};
exports.global = global;
var truncate = {
  end: 'bx--text-truncate--end'
};
exports.truncate = truncate;
var sidenav = {
  sidenav: 'bx--side-nav',
  state: {
    expanded: 'bx--side-nav--expanded'
  },
  variants: {
    fixed: 'bx--side-nav--fixed'
  }
};
exports.sidenav = sidenav;
