// carbon-components-react/lib/components/UIShell/HeaderMenuItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classNames = require("./classNames");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var HeaderMenuItem = function HeaderMenuItem(props) {
  var className = props.className,
      children = props.children,
      role = props.role,
      innerRef = props.innerRef,
      rest = _objectWithoutProperties(props, ["className", "children", "role", "innerRef"]);

  return _react.default.createElement("li", {
    className: className,
    role: role
  }, _react.default.createElement("a", _extends({}, rest, {
    className: _classNames.navigation.menuitem,
    ref: innerRef,
    role: "menuitem"
  }), _react.default.createElement("span", {
    className: _classNames.truncate.end
  }, children)));
};

HeaderMenuItem.propTypes = {
  /**
   * Optionally provide a custom class to apply to the underlying <li> node
   */
  className: _propTypes.default.string,

  /**
   * Pass in children that are either a string or can be read as a string by
   * screen readers
   */
  children: _propTypes.default.node.isRequired,

  /**
   * Optionally supply a role for the underlying <li> node. Useful for resetting
   * <ul> semantics for menus.
   */
  role: _propTypes.default.string
};

var _default = _react.default.forwardRef(function (props, ref) {
  return _react.default.createElement(HeaderMenuItem, _extends({}, props, {
    innerRef: ref
  }));
});

exports.default = _default;
