// carbon-components-react/lib/components/UIShell/SideNavFooter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// TODO: replace with @carbon/icons-react
var Close = function Close() {
  return _react.default.createElement("svg", {
    "aria-hidden": "true",
    width: "20",
    height: "20",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 32 32"
  }, _react.default.createElement("path", {
    d: "M17.414 16L24 9.414 22.586 8 16 14.586 9.414 8 8 9.414 14.586 16 8 22.586 9.414 24 16 17.414 22.586 24 24 22.586 17.414 16z"
  }));
}; // TODO: replace with @carbon/icons-react


var ChevronRight = function ChevronRight() {
  return _react.default.createElement("svg", {
    "aria-hidden": "true",
    width: "20",
    height: "20",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 32 32"
  }, _react.default.createElement("path", {
    d: "M22 16L12 26l-1.414-1.414L19.172 16l-8.586-8.586L12 6l10 10z"
  }));
};
/**
 * SideNavFooter is used for rendering the button at the bottom of the side
 * navigation that is a part of the UI Shell. It is responsible for handling the
 * user interaction to expand or collapse the side navigation.
 */


var SideNavFooter = function SideNavFooter(_ref) {
  var assistiveText = _ref.assistiveText,
      customClassName = _ref.className,
      isExpanded = _ref.isExpanded,
      onToggle = _ref.onToggle;
  var className = (0, _classnames.default)('bx--side-nav__footer', customClassName);
  return _react.default.createElement("footer", {
    className: className
  }, _react.default.createElement("button", {
    className: "bx--side-nav__toggle",
    type: "button",
    onClick: onToggle,
    title: assistiveText
  }, _react.default.createElement("div", {
    className: "bx--side-nav__icon"
  }, isExpanded ? _react.default.createElement(Close, null) : _react.default.createElement(ChevronRight, null)), _react.default.createElement("span", {
    className: "bx--assistive-text"
  }, assistiveText)));
};

SideNavFooter.propTypes = {
  /**
   * Provide text to be read to screen readers and shown as a tooltip when
   * interacting with the toggle button in the footer
   */
  assistiveText: _propTypes.default.string.isRequired,

  /**
   * Specify whether the side navigation is expanded or collapsed
   */
  isExpanded: _propTypes.default.bool.isRequired,

  /**
   * Provide a function that is called when the toggle button is interacted
   * with. Useful for controlling the expansion state of the side navigation.
   */
  onToggle: _propTypes.default.func.isRequired
};
SideNavFooter.defaultProps = {
  assistiveText: 'Toggle opening or closing the side navigation'
};
var _default = SideNavFooter;
exports.default = _default;
