// carbon-components/umd/components/radio-button/radio-button.config.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['module', '../../globals/js/settings'], factory);
  } else if (typeof exports !== "undefined") {
    factory(module, require('../../globals/js/settings'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod, global.settings);
    global.radioButtonConfig = mod.exports;
  }
})(this, function (module, _require) {
  'use strict';

  var prefix = _require.prefix;


  var items = [{
    id: 'radio-button-1',
    value: 'red',
    label: 'Radio Button label'
  }, {
    id: 'radio-button-2',
    value: 'green',
    label: 'Radio Button label'
  }, {
    id: 'radio-button-3',
    value: 'blue',
    label: 'Radio Button label',
    disabled: true
  }];

  module.exports = {
    context: {
      prefix: prefix
    },
    variants: [{
      name: 'default',
      label: 'Progress Indicator',
      context: {
        selectedValue: 'red',
        group: 'radio-button',
        items: items
      }
    }]
  };
});
