// carbon-components/umd/components/form/form.config.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['module', '../../globals/js/settings'], factory);
  } else if (typeof exports !== "undefined") {
    factory(module, require('../../globals/js/settings'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod, global.settings);
    global.formConfig = mod.exports;
  }
})(this, function (module, _require) {
  'use strict';

  var prefix = _require.prefix;


  var selectItems = [{
    label: 'Choose an option',
    disabled: true,
    selected: true,
    hidden: true
  }, {
    label: 'A much longer option that is worth having around to check how text flows',
    value: 'solong'
  }, {
    label: 'Category 1',
    items: [{
      label: 'Option 1',
      value: 'option1'
    }, {
      label: 'Option 2',
      value: 'option2'
    }]
  }, {
    label: 'Category 2',
    items: [{
      label: 'Option 1',
      value: 'option1'
    }, {
      label: 'Option 2',
      value: 'option2'
    }]
  }];

  module.exports = {
    context: {
      prefix: prefix
    },
    variants: [{
      name: 'default',
      label: 'Form',
      notes: 'Forms are widely used to collect user input.',
      context: {
        selectItems: selectItems
      }
    }, {
      name: 'light',
      label: 'Form (Light)',
      context: {
        light: true,
        selectItems: selectItems
      }
    }]
  };
});
