// carbon-components/umd/components/loading/loading.config.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['module', '../../globals/js/settings', '../../globals/js/feature-flags'], factory);
  } else if (typeof exports !== "undefined") {
    factory(module, require('../../globals/js/settings'), require('../../globals/js/feature-flags'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod, global.settings, global.featureFlags);
    global.loadingConfig = mod.exports;
  }
})(this, function (module, _require, _require2) {
  'use strict';

  var prefix = _require.prefix;
  var componentsX = _require2.componentsX;


  module.exports = {
    context: {
      prefix: prefix,
      componentsX: componentsX
    },
    variants: [{
      name: 'default',
      label: 'Loading',
      notes: '\n        Loading spinners are used when retrieving data or performing slow computations,\n        and help to notify users that loading is underway.\n      ',
      context: {
        overlay: true
      }
    }, {
      name: 'without-overlay',
      label: 'Without overlay',
      context: {
        overlay: false
      }
    }, {
      name: 'small',
      label: 'Small',
      context: {
        overlay: false,
        small: true
      }
    }]
  };
});
