// carbon-components/umd/components/tag/tag.config.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['module', '../../globals/js/settings', '../../globals/js/feature-flags'], factory);
  } else if (typeof exports !== "undefined") {
    factory(module, require('../../globals/js/settings'), require('../../globals/js/feature-flags'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod, global.settings, global.featureFlags);
    global.tagConfig = mod.exports;
  }
})(this, function (module, _require, _require2) {
  'use strict';

  var prefix = _require.prefix;
  var componentsX = _require2.componentsX;


  var tags = !componentsX ? [{
    type: 'ibm',
    label: 'IBM'
  }, {
    type: 'beta',
    label: 'Beta'
  }, {
    type: 'third-party',
    label: 'Third-Party'
  }, {
    type: 'local',
    label: 'Local'
  }, {
    type: 'dedicated',
    label: 'Dedicated'
  }, {
    type: 'custom',
    label: 'Custom'
  }, {
    type: 'experimental',
    label: 'Experimental'
  }, {
    type: 'community',
    label: 'Community'
  }, {
    type: 'private',
    label: 'Private'
  }] : [{
    type: 'basic',
    label: 'Component'
  }, {
    type: 'red',
    label: 'Red'
  }, {
    type: 'magenta',
    label: 'Magenta'
  }, {
    type: 'purple',
    label: 'Purple'
  }, {
    type: 'blue',
    label: 'Blue'
  }, {
    type: 'cyan',
    label: 'Cyan'
  }, {
    type: 'teal',
    label: 'Teal'
  }, {
    type: 'green',
    label: 'Green'
  }, {
    type: 'cool-gray',
    label: 'Cool-Gray'
  }, {
    type: 'warm-gray',
    label: 'Warm-Gray'
  }];

  module.exports = {
    context: {
      prefix: prefix
    },
    variants: [{
      name: 'default',
      label: 'Tag',
      context: {
        tags: tags,
        componentsX: componentsX
      }
    }]
  };
});
