// carbon-components/umd/components/modal/modal.config.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['module', '../../globals/js/settings'], factory);
  } else if (typeof exports !== "undefined") {
    factory(module, require('../../globals/js/settings'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod, global.settings);
    global.modalConfig = mod.exports;
  }
})(this, function (module, _require) {
  'use strict';

  var prefix = _require.prefix;


  module.exports = {
    context: {
      prefix: prefix
    },
    variants: [{
      name: 'default',
      label: 'Transactional Modal',
      notes: '\n        Modals communicate information via a secondary window and allow the user to maintain the context of a particular task.\n      ',
      context: {
        idSuffix: Math.random().toString(36).substr(2),
        hasFooter: true,
        classPrimaryButton: prefix + '--btn--primary',
        classCloseButton: prefix + '--btn--secondary'
      }
    }, {
      name: 'nofooter',
      label: 'Passive Modal',
      notes: 'Passive Modals are modals without footers.',
      context: {
        idSuffix: Math.random().toString(36).substr(2),
        hasFooter: false,
        classPrimaryButton: prefix + '--btn--primary',
        classCloseButton: prefix + '--btn--secondary'
      }
    }, {
      name: 'danger',
      label: 'Danger Modal',
      context: {
        idSuffix: Math.random().toString(36).substr(2),
        hasFooter: true,
        labelPrimaryButton: 'Danger',
        classModalSupplemental: prefix + '--modal--danger',
        classPrimaryButton: prefix + '--btn--danger--primary',
        classCloseButton: prefix + '--btn--tertiary'
      }
    }, {
      name: 'input',
      label: 'Input Modal',
      context: {
        idSuffix: Math.random().toString(36).substr(2),
        hasInput: true,
        hasFooter: true,
        classPrimaryButton: prefix + '--btn--primary',
        classCloseButton: prefix + '--btn--secondary'
      }
    }]
  };
});
